src/pocsag/pocsag.cpp

```diff
@@ -465,154 +491,160 @@ size_t numericMessageLength(int address, int numChars) {
-    
-	unsigned char *TabSymbol=(unsigned char *)malloc(Size*32);
-	int Sym=0;
-	
-	for(int i=0;i<Size;i++)
+
+    float Deviation = 4500;
+    int FiFoSize = 12000;
+    if (debug)
+        fprintf(stderr, "Fifo Size = %d, Size = %d, Baud rate = %d\n", FiFoSize, Size, SR);
+    fskburst fsktest(Freq - Deviation, SR, Deviation * 2, 14, FiFoSize, 1, 0.0);
+
+    unsigned char *TabSymbol = (unsigned char *)malloc(Size * 32);
+    int Sym = 0;
+
+    for (int i = 0; i < Size; i++)
+    {
+        if (!Inverted)
+            Message[i] = ~Message[i];
+        for (int j = 31; j >= 0; j--)
         {
-            if(!Inverted) Message[i] = ~Message[i];
-            for(int j=31;j>=0;j--)
+            TabSymbol[Sym] = (Message[i] >> j) & 0x1;
+            if (debug)
             {
-                TabSymbol[Sym]=(Message[i]>>j)&0x1;
-                if(debug)
-                {
-                  fprintf(stderr,"%x",TabSymbol[Sym]);
-                  if (j==16) fprintf(stderr," ");
-                }
-                Sym++;
+                fprintf(stderr, "%x", TabSymbol[Sym]);
+                if (j == 16)
+                    fprintf(stderr, " ");
             }
-            if(debug) fprintf(stderr,"\n");
+            Sym++;
         }
-        if(debug) fprintf(stderr,"Symbols=%d\n",Sym);
-		fsktest.SetSymbols(TabSymbol,Sym);
-		
-		/*for(i=0;i<FiFoSize;i++)
+        if (debug)
+            fprintf(stderr, "\n");
+    }
+    if (debug)
+        fprintf(stderr, "Symbols=%d\n", Sym);
+    fsktest.SetSymbols(TabSymbol, Sym);
+
+    /*for(i=0;i<FiFoSize;i++)
 		{
 			TabSymbol[i]=1;
 		}	
 		fsktest.SetSymbols(TabSymbol,FiFoSize);
 		sleep(1);*/
-		
-		
-	
-	fsktest.stop();
+
+    fsktest.stop();
 }
 
 void print_usage(void)
 {
 
-fprintf(stderr,\
-"\npocsag -%s\n\
+    fprintf(stderr,
+            "\npocsag -%s\n\
 Usage:\npocsag  [-f Frequency] [-i] [-r Rate]\n\
 -f float      central frequency Hz(50 kHz to 1500 MHz),\n\
 -r int        baud rate (512, 1200 or 2400. Default 1200 bps),\n\
 -b int        function bits (0-3. Default 3),\n\
 -n            use numeric messages,\n\
+-t int        repeat messages X times (Default 4)\n\
 -i            invert the modulation polarity,\n\
 -d            debug,\n\
 -?            help (this help).\n\
-\n",\
-PROGRAM_VERSION);
+\n",
+            PROGRAM_VERSION);
 
 } /* end function print_usage */
 
-int main(int argc, char* argv[]) {
+int main(int argc, char *argv[]) {
     //Read in lines from STDIN.
     //Lines are in the format of address:message
     //The program will encode transmissions for each message, writing them
     //to STDOUT. It will also encode a rand amount of silence between them,
     //from 1-10 seconds in length, to act as a simulated "delay".
     int a;
     int anyargs = 1;
-    uint64_t SetFrequency=466230000L;
+    uint64_t SetFrequency = 466230000L;
     int SetRate = 1200;
     int SetFunctionBits = 3;
+    int REPEAT_COUNT = 4;
```

Comment:
I would suggest to set default for repeat count to 1. While pagers can de-duplicate messages, and suppress alerts on duplicates, in case same message is being transmitted from multiple towers, this is a pager configuration option and, by default, you will get 4 alerts.

`  For wide area coverage some systems use multiple transmitters located
  throughout the coverage area. By transmitting the pager information
  sequentially from one station and then the next and so on, the likelihood
  that the pager will receive the information is high. In such a system
  a pager may receive the same information more than once and alert for
  each time. By setting a sequential lockout time, the pager will ignore
  duplicate transmissions within that time period and will not alert for
  each transmission.`

Reply:
im fine with that